Dockerfile

```diff
@@ -0,0 +1,56 @@
+FROM --platform=x86_64 debian:bookworm AS firmware-builder
+
+# FIRMWARE_GIT should be a git repo with the firmware source code
+ARG FIRMWARE_GIT=https://github.com/flipperdevices/flipperzero-firmware.git
+
+# Install dependencies
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    ca-certificates \
+    git \
+    curl \
+    clang \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /app
+
+RUN git clone --recursive $FIRMWARE_GIT firmware
+
+WORKDIR /app/firmware
+
+RUN ./fbt
+
+####################
+# bindgen
+FROM --platform=x86_64 rust:bookworm AS builder
+
+# Install dependencies
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    ca-certificates \
+    libudev-dev \
+    libclang-dev \
+    clang \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /app
+
+COPY --from=firmware-builder /app/firmware /app/firmware
+
+WORKDIR /app/flipperzero-rs/
+
+COPY . .
+
+WORKDIR /app/flipperzero-rs/tools
+
+RUN cargo run --bin generate-bindings /app/firmware/build/f7-firmware-D/sdk_headers/
+
+RUN cp bindings.rs ../crates/sys/src
+
+WORKDIR /app/flipperzero-rs/crates
+
+# test it builds
+RUN cargo build --release
+
+####################
+ENTRYPOINT [ "cat", "sys/src/bindings.rs" ]
```

Comment:
The generated bindings having no formatting whatsoever, which means that the usage steps are incomplete and multi-step. Is the problem that the generated bindings inside Docker are not formatted (in which case the Dockerfile needs a `cargo fmt` step), or is it a problem with this method of extracting them?

Reply:
will get it formatted in the Dockerfile.